packages/types/app/vue.d.ts

```diff
@@ -16,12 +21,150 @@ type DefaultData<V> = object | ((this: V) => object)
 type DefaultProps = Record<string, any>
 type DefaultMethods<V> = { [key: string]: (this: V, ...args: any[]) => any }
 type DefaultComputed = { [key: string]: any }
-type DefaultAsyncData<V> = ((this: V, context: Context) => Promise<object | void> | object | void)
+type DefaultAsyncData = ((context: Context) => Promise<object | void> | object | void)
```

Comment:
I think we need to explicitly set to never or it will still exist.

```suggestion
type DefaultAsyncData = ((this: never, context: Context) => Promise<object | void> | object | void)
```